### DISCUSSION TITLE
gg: Support for Joysticks.

### DISCUSSION THREAD

**Discussion by KeitoTobi1** [vlang/v#11906] [Ideas]
I'm making a game with `gg`, but I thought it would be inconvenient if I couldn't use a joystick when making an indie game, so I issued this issue.

**Comment by larpon** [comment]
If you need it here and now I think one of the V SDL wraps should be able to help you 🙂

**Comment by KeitoTobi1** [comment]
However, the V SDL Warps project update seems to be stalled.

**Comment by larpon** [comment]
> However, the V SDL Warps project update seems to be stalled.

It's a wrapper project - I don't expect it to be very active.

Here's the ones that I can think of:
https://github.com/vlang/sdl
https://github.com/kjlaw89/vsdl
https://github.com/nsauzede/vsdl2

**Comment by KeitoTobi1** [comment]
I'm talking about https://github.com/vlang/sdl.

**Comment by JalonSolov** [comment]
Yeah, that one definitely needs to be updated, due to changes in V.

**Comment by spytheman** [comment]
With latest V and some cosmetic changes to https://github.com/nsauzede/vsdl2/ , the tvintris example there compiles cleanly and works.
I've made a PR with the changes: https://github.com/nsauzede/vsdl2/pull/10 .

**Comment by spytheman** [comment]
![image](https://user-images.githubusercontent.com/26967/128312418-e4870241-f65f-498b-aae5-9728a578a47f.png)

**Comment by spytheman** [comment]
Note that <https://github.com/vlang/sdl> is/was a fork of <https://github.com/nsauzede/vsdl2> - it initially was copied in the main V repo under `vlib/sdl`, and then split to <https://github.com/vlang/sdl>, after we migrated gg from using glfw to sokol.

**Comment by spytheman** [comment]
I've just made another PR too: https://github.com/kjlaw89/vsdl/pull/4

**Comment by KeitoTobi1** [comment]
I sent a another PR to https://github.com/vlang/sdl, what should I do?

**Comment by larpon** [comment]
> I sent a another PR to https://github.com/vlang/sdl, what should I do?

@medvednikov when you have time to merge ^
